qemu/tests/mq_enabled_chk.py

```diff
@@ -0,0 +1,114 @@
+                err_msg = "Param is invalid, "
+            err_msg += "Current queues value is not expected, "
+            expect_q_number = queues_status[1]
+
+        if len(err_msg) > 0:
+            err_msg += "current queues set is %s, " % cur_queues_status[1]
+            err_msg += "max allow queues set is %s, " % cur_queues_status[0]
+            err_msg += "when run cmd: '%s', " % mq_set_cmd
+            err_msg += "expect queues are %s," % expect_q_number
+            err_msg += "expect max allow queues are %s, " % queues_status[0]
+            err_msg += "output: '%s'" % output
+            test.fail(err_msg)
+
+        return [int(_) for _ in cur_queues_status]
+
+    def get_queues_status(session, ifname, timeout=240):
+        """
+        Get queues status
+        """
+        mq_get_cmd = "ethtool -l %s" % ifname
+        nic_mq_info = session.cmd_output(mq_get_cmd, timeout=timeout,
+                                         safe=True)
+        queues_reg = re.compile(r"Combined:\s+(\d)", re.I)
+        queues_info = queues_reg.findall(" ".join(nic_mq_info.splitlines()))
+        if len(queues_info) != 2:
+            err_msg = "Oops, get guest queues info failed, "
+            err_msg += "make sure your guest support MQ.\n"
+            err_msg += "Check cmd is: '%s', " % mq_get_cmd
+            err_msg += "Command output is: '%s'." % nic_mq_info
+            test.cancel(err_msg)
+        return [int(x) for x in queues_info]
+
+    def chk_mq_enabled(vm, queues):
+        login_timeout = int(params.get("login_timeout", 360))
+        session = vm.wait_for_login(timeout=login_timeout)
+
+        mac = vm.get_mac_address(0)
+        ifname = utils_net.get_linux_ifname(session, mac)
+        queues_status = get_queues_status(session, ifname)
+
+        session.close()
+        if not queues_status[0] == queues or \
+                not queues_status[1] == min(queues, int(vm.cpuinfo.smp)):
+            txt = "Pre-set maximums Combined should equals to queues in qemu"
+            txt += " cmd line.\n"
+            txt += "Current hardware settings Combined should be the min of "
+            txt += "queues and smp.\n"
+            txt += "Pre-set maximum Combined is: %s, " % queues_status[0]
+            txt += " queues in qemu cmd line is: %s.\n" % queues
+            txt += "Current hardware settings Combined "
+            txt += "is: %s, " % queues_status[1]
+            txt += " smp in qemu cmd line is: %s." % int(vm.cpuinfo.smp)
+            test.fail(txt)
+
+    error_context.context("Init the guest and try to login", logging.info)
+    queues_list = params["queues_list"].split()
+
+    for queues in queues_list:
+        params["queues"] = int(queues)
+        params["start_vm"] = "yes"
+
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+
+        chk_mq_enabled(vm, int(queues))
+        change_queues_number(vm, 2)
```

Comment:
Should be changing same value as queues number, not a fixed number '2' ?

Reply:
A bit more info: The test case requires setting combined value same as queues value.